src/ui/ui_apps/ui_apps_mixin.test.js

```diff
@@ -0,0 +1,100 @@
+import { uiAppsMixin } from './ui_apps_mixin';
+
+jest.mock('./ui_app', () => ({
+  UiApp: class StubUiApp {
+    constructor(kbnServer, spec) {
+      this._id = spec.id;
+      this._hidden = !!spec.hidden;
+    }
+    getId() {
+      return this._id;
+    }
+    isHidden() {
+      return this._hidden;
+    }
+  }
+}));
+
+describe('UiAppsMixin', () => {
+  let kbnServer;
+  let server;
+
+  beforeAll(() => {
+    kbnServer = {
+      uiExports: {
+        uiAppSpecs: [
+          {
+            id: 'foo',
+            hidden: true,
+          },
+          {
+            id: 'bar',
+            hidden: false,
+          },
+        ]
+      }
+    };
+
+    server = {
+      decorate: jest.fn((type, name, value) => {
+        if (type !== 'server') {
+          return;
+        }
+
+        server[name] = value;
+      }),
+    };
+
+    uiAppsMixin(kbnServer, server);
+  });
+
+  it('creates kbnServer.uiApps from uiExports', () => {
+    expect(kbnServer.uiApps).toMatchSnapshot();
+  });
+
+  it('decorates server with methods', () => {
+    expect(server.decorate.mock.calls).toMatchSnapshot();
+  });
+
+  describe('server.getAllUiApps()', () => {
+    it('returns hidden and non-hidden apps', () => {
+      expect(server.getAllUiApps()).toMatchSnapshot();
+    });
+  });
+
+  describe('server.getUiAppById()', () => {
+    it('returns non-hidden apps when requested, undefined for non-hidden and unknown apps', () => {
+      expect(server.getUiAppById('foo')).toMatchSnapshot('hidden');
```

Comment:
__optional nit:__ I don't have a strong opinion on that so up to you, but using snapshot just for `undefined` (here and in test below) sounds like an overkill to me.

Reply:
🤷‍♀️ I like consistency

Reply:
Nit: I think this should just be inlined as `undefined`. Same with the `{}` ones. I think we should optimize for readability, not consistency.